src/libraries/System.Net.Quic/ref/System.Net.Quic.cs

```diff
@@ -52,6 +52,7 @@ public static partial class QuicImplementationProviders
     }
     public sealed partial class QuicListener : System.IDisposable
     {
+        public delegate System.Threading.Tasks.ValueTask<System.Net.Security.SslServerAuthenticationOptions> ServerOptionsSelectionCallback(object sender, System.Net.Security.SslClientHelloInfo clientHelloInfo, object? state, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken));
```

Comment:
Seems like this should be on the Options class, not on the Listener itself. Is there a reason it isn't?

Reply:
Or actually, why is this delegate definition part of QUIC at all? It doesn't seem to have any QUIC-specific arguments. How is this different than whatever we do for SslStream today?

Reply:
In the existing delegate the sender is SslStream. That does not seems appropriate for Quic. Here the sender should probably be the original `QuicListener` but I'm not sure we have access to it. So for now, I left it as `object`.

Reply:
it is gone now.